drivers/display/display_ili9341.h

```diff
@@ -0,0 +1,109 @@
+#define ILI9341_VMCTRL2_LEN 1U
+#define ILI9341_PGAMCTRL_LEN 15U
+#define ILI9341_NGAMCTRL_LEN 15U
+
+#define ILI9341_PWCTRLA_LEN 5U
+#define ILI9341_PWCTRLB_LEN 3U
+#define ILI9341_PWSEQCTRL_LEN 4U
+#define ILI9341_TIMCTRLA_LEN 3U
+#define ILI9341_TIMCTRLB_LEN 2U
+#define ILI9341_PUMPRATIOCTRL_LEN 1U
+#define ILI9341_EN3G_LEN 1U
+#define ILI9341_ETMOD_LEN 1U
+
+/** X resolution (pixels). */
+#define ILI9341_X_RES 240U
+/** Y resolution (pixels). */
+#define ILI9341_Y_RES 320U
+
+/** ILI9341 registers to be initialized. */
+struct ili9341_regs {
+	uint8_t gamset[ILI9341_GAMSET_LEN];
+	uint8_t frmctr1[ILI9341_FRMCTR1_LEN];
+	uint8_t disctrl[ILI9341_DISCTRL_LEN];
+	uint8_t pwctrl1[ILI9341_PWCTRL1_LEN];
+	uint8_t pwctrl2[ILI9341_PWCTRL2_LEN];
+	uint8_t vmctrl1[ILI9341_VMCTRL1_LEN];
+	uint8_t vmctrl2[ILI9341_VMCTRL2_LEN];
+	uint8_t pgamctrl[ILI9341_PGAMCTRL_LEN];
+	uint8_t ngamctrl[ILI9341_NGAMCTRL_LEN];
+
+	uint8_t pwctrla[ILI9341_PWCTRLA_LEN];
+	uint8_t pwctrlb[ILI9341_PWCTRLB_LEN];
+	uint8_t pwseqctrl[ILI9341_PWSEQCTRL_LEN];
+	uint8_t timctrla[ILI9341_TIMCTRLA_LEN];
+	uint8_t timctrlb[ILI9341_TIMCTRLB_LEN];
+	uint8_t pumpratio[ILI9341_PUMPRATIOCTRL_LEN];
+	uint8_t en3g[ILI9341_EN3G_LEN];
+	uint8_t entrymdset[ILI9341_ETMOD_LEN];
+};
+
+/* Initializer macro for ILI9341 registers. */
+#define ILI9341_REGS_INIT(n)                                                                       \
+	static const struct ili9341_regs ili9xxx_regs_##n = {                                      \
+		.gamset = DT_PROP(DT_INST(n, ilitek_ili9341), gamset),                             \
+		.frmctr1 = DT_PROP(DT_INST(n, ilitek_ili9341), frmctr1),                           \
+		.disctrl = DT_PROP(DT_INST(n, ilitek_ili9341), disctrl),                           \
+		.pwctrl1 = DT_PROP(DT_INST(n, ilitek_ili9341), pwctrl1),                           \
+		.pwctrl2 = DT_PROP(DT_INST(n, ilitek_ili9341), pwctrl2),                           \
+		.vmctrl1 = DT_PROP(DT_INST(n, ilitek_ili9341), vmctrl1),                           \
+		.vmctrl2 = DT_PROP(DT_INST(n, ilitek_ili9341), vmctrl2),                           \
+		.pgamctrl = DT_PROP(DT_INST(n, ilitek_ili9341), pgamctrl),                         \
+		.ngamctrl = DT_PROP(DT_INST(n, ilitek_ili9341), ngamctrl),                         \
+		.pwctrla = DT_PROP(DT_INST(n, ilitek_ili9341), pwctrla),                           \
+		.pwctrlb = DT_PROP(DT_INST(n, ilitek_ili9341), pwctrlb),                           \
+		.pwseqctrl = DT_PROP(DT_INST(n, ilitek_ili9341), pwseqctrl),                       \
+		.timctrla = DT_PROP(DT_INST(n, ilitek_ili9341), timctrla),                         \
+		.timctrlb = DT_PROP(DT_INST(n, ilitek_ili9341), timctrlb),                         \
+		.pumpratio = DT_PROP(DT_INST(n, ilitek_ili9341), pumpratio),                       \
+		.en3g = DT_PROP(DT_INST(n, ilitek_ili9341), en3g),                                 \
+		.entrymdset = DT_PROP(DT_INST(n, ilitek_ili9341), entrymdset),                     \
+	}
+
```

Comment:
It would be good to add some build time asserts that the array length for each register is correct.

Reply:
transmit_regs - is created on the stack and does not increase the RAM usage in the main program
```
	const struct ili9xxx_config *config = dev->config;
	const struct ili9341_regs *regs = config->regs;
	int r;
	const struct {
		const uint8_t *reg;
		unsigned int reg_len;
		const char *reg_name;
		uint8_t reg_id;
	} transmit_regs[] = {
		{ NULL, 0, "SW RESET", ILI9341_SWRESET },
		{ NULL, 0, "DISPOFF", ILI9XXX_DISPOFF },
		{ regs->pwseqctrl, ILI9341_PWSEQCTRL_LEN, "PWSEQCTRL", ILI9341_PWSEQCTRL },
		{ regs->timctrla, ILI9341_TIMCTRLA_LEN, "TIMCTRLA", ILI9341_TIMCTRLA },
		{ regs->timctrlb, ILI9341_TIMCTRLB_LEN, "TIMCTRLB", ILI9341_TIMCTRLB },
		{ regs->pwctrla, ILI9341_PWCTRLA_LEN, "PWCTRLA", ILI9341_PWCTRLA },
		{ regs->pwctrlb, ILI9341_PWCTRLB_LEN, "PWCTRLB", ILI9341_PWCTRLB },
		{ regs->pumpratio, ILI9341_PUMPRATIOCTRL_LEN, "PUMPRATIOCTRL",
		  ILI9341_PUMPRATIOCTRL },
		{ regs->gamset, ILI9341_GAMSET_LEN, "GAMSET", ILI9341_GAMSET },
		{ regs->frmctr1, ILI9341_FRMCTR1_LEN, "FRMCTR1", ILI9341_FRMCTR1 },
		{ regs->disctrl, ILI9341_DISCTRL_LEN, "DISCTRL", ILI9341_DISCTRL },
		{ regs->pwctrl1, ILI9341_PWCTRL1_LEN, "PWCTRL1", ILI9341_PWCTRL1 },
		{ regs->pwctrl2, ILI9341_PWCTRL2_LEN, "PWCTRL2", ILI9341_PWCTRL2 },
		{ regs->vmctrl1, ILI9341_VMCTRL1_LEN, "VMCTRL1", ILI9341_VMCTRL1 },
		{ regs->vmctrl2, ILI9341_VMCTRL2_LEN, "VMCTRL2", ILI9341_VMCTRL2 },
		{ regs->pgamctrl, ILI9341_PGAMCTRL_LEN, "PGAMCTRL", ILI9341_PGAMCTRL },
		{ regs->ngamctrl, ILI9341_NGAMCTRL_LEN, "NGAMCTRL", ILI9341_NGAMCTRL },
		{ regs->en3g, ILI9341_EN3G_LEN, "EN3G", ILI9341_EN3G },
		{ regs->entrymdset, ILI9341_ETMOD_LEN, "ETMOD", ILI9341_ETMOD },
	};

	for (int i = 0; i < ARRAY_SIZE(transmit_regs); i++) {
		LOG_HEXDUMP_DBG(transmit_regs[i].reg, transmit_regs[i].reg_len,
				transmit_regs[i].reg_name);

		r = ili9xxx_transmit(dev, transmit_regs[i].reg_id, transmit_regs[i].reg,
				     transmit_regs[i].reg_len);
		if (r < 0) {
			return r;
		}
		if (i == 0) {
			k_sleep(K_MSEC(5));
		}
	}
```

Reply:
When compiling in debug mode with LOG_INF enabled (without outputting messages to the HEX message log), the loop function variant showed a code of 160 bytes more FLASH usage. And, in my opinion, function code without a loop should use less RAM. 
So far, I leave the code version without a loop.